jax/_src/lax/control_flow/for_loop.py

```diff
@@ -138,6 +141,16 @@ def for_loop(nsteps, body, init_state):
   Returns:
     A Pytree of values representing the output of the for loop.
   """
+  if isinstance(nsteps, int):
```

Comment:
Note to future selves: this is a slick way to generate nested loops at the api level (w/o making the primitive more complex), but we're not sure if keeping the loops flattened in the lowering could be advantageous, and we're also not sure if that's hard to recover from nested loops.